proposed/2022/package-scoring-v1.md

```diff
@@ -0,0 +1,222 @@
+
+EX: A package is in control of providing ample metadata to ensure the quality of the package reaches the full score. A package is not in control of how popular it may be however.
+
+To scale package scoring for the future, each score will be comprised of an analysis of package performance and potential package/dependency issues. Each of these issues will be categorized into one of these four categories and empower the author and consumer to resolve these issues to the best of their ability and control.
+
+#### Issues
+
+As a means to scale package scoring to it's potential through continuous iteration, each category will have its own set of issues that a user can be aware of when browsing through the package. Keeping the same idea of **being in control** or **not in control** also applies to package issues.
+
+For issues which can be addressed, an issue will include an empowering message regarding how one can take action to help address the issue.
+
+EX: A package might be missing a README which a developer may not be able to easily get started with the package. They might be able to then contribute a new issue or even the inclusion of the README file to the NuGet package if it's open source.
+
+**Issue Examples:**
+
+- Invalid .nuspec / .csproj
+- Missing dependency
+- Missing README
+- No example
+- Not enough public API documentation
+- Bad semver
+- Not v1+
+- No website
+- No repository
+- Unmaintained
+- No bug tracker
+- Has CVE (Critical, High, etc)
+- Deprecated license
+- Missing license
+- Non SPDX license
+- Unsafe copyright
+- Deprecated
+- Empty package
+- Malware
+- etc
+
+#### Popularity
+
+Popularity is the more unique of the other three categories which are more issue based. Popularity is calculated as how many projects depend on a package over the past 30 days. This is scored as a percentile up to 100% (most used) to 0% (least used).
+
+Since this is not possible with current data pipelines, we will work on providing such an experience in the future and use two proxy values such as total weekly downloads and total count of packages depending on the package to be scored up to a percentile of up to 100%.
+
+- Total Weekly Downloads
+- Number of Dependents
+
+Total Score - 100
+
+#### Quality
+
+Quality is the combination of:
+
+- Follow NuGet conventions
+  - Provides a valid .csproj / .nuspec
+  - Provides a valid README
+- Provide documentation
+  - Provides an example
+  - 20% or more of the public API has xml doc comments
+- Platform support
+  - Supports all possible modern platforms i.e. [.NET 5+/.NET Standard](https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#net-5-and-net-standard) (iOS, Android, Windows, MacOS, Linux, etc)
+- Pass static analysis
+  - Code has no errors, warnings, lints, or formatting issues.
+- Support up-to-date dependencies
+  - All of the package dependencies are supported in the latest version
```

Comment:
Not sure I understand this. 
Because of how compatibility works, if a package supports a certain framework, then all future versions of that framework are likely to be supported as well. 

If a package claims to support a framework but their dependencies don't then that's a problem package. 
Maybe that's what we should focus on instead, ensure that the package is installable in all of it's declared frameworks.